src/Asset/AssetManager.php

```diff
@@ -584,76 +849,74 @@ protected function publishDirectory($src, $options)
                     };
                 }
             }
+
             if (!isset($opts['afterCopy']) && $this->afterCopy !== null) {
                 $opts['afterCopy'] = $this->afterCopy;
             }
+
             FileHelper::copyDirectory($src, $dstDir, $opts);
         }
 
+
         return [$dstDir, $this->baseUrl.'/'.$dir];
     }
 
     /**
-     * Returns the published path of a file path.
-     * This method does not perform any publishing. It merely tells you
-     * if the file or directory is published, where it will go.
-     *
-     * @param string $path directory or file path being published
+     * @param AssetBundle $bundle
+     * @param string      $asset
      *
-     * @return string|false string the published file path. False if the file or directory does not exist
+     * @return string|bool
      */
-    public function getPublishedPath($path)
+    protected function resolveAsset(AssetBundle $bundle, string $asset)
     {
-        $path = $this->app->getAlias($path);
+        if (isset($this->assetMap[$asset])) {
+            return $this->assetMap[$asset];
+        }
 
-        if (isset($this->_published[$path])) {
-            return $this->_published[$path][0];
+        if ($bundle->sourcePath !== null && Info::isRelative($asset)) {
```

Comment:
I haven't mean putting `isRelative` into `Info`. It could be made a private method.